awsmp/_driver.py

```diff
@@ -165,6 +167,28 @@ def get_entity_details(entity_id: str) -> Dict:
     return json.loads(e["Details"])
 
 
+def get_public_offer_id(entity_id: str):
+    client = get_client()
+    e = client.list_entities(
+        Catalog="AWSMarketplace",
+        EntityType="Offer",
+        EntityTypeFilters={
+            "OfferFilters": {
+                "ProductId": {
+                    "ValueList": [
+                        entity_id,
+                    ]
+                },
+                "Targeting": {"ValueList": ["None"]},
+            }
+        },
+    )
+    if not e["EntitySummaryList"]:
+        raise ResourceNotFoundException
```

Comment:
would be useful here to know which resource was not found. So:
```
raise ResourceNotFoundException(f"Offer with entity-id {entity_id} not found")

Reply:
I added logger for better information to user.

Reply:
don't use `logger.exception()` outside from an exception handler. See https://docs.python.org/3/library/logging.html#logging.Logger.exception